project/scratch.py

```diff
@@ -6,32 +6,49 @@
 
 # a.sort()
 #binary sory.
-def binary_search(arr, target): 
-    # step 1. we find the midpoint.
-    midpoint = len(arr) // 2
-    while len(arr) > 3:
-    # We check to see if midpoint is equal to target.
-        if target == arr[midpoint]:
-            return target
-        # we check to see which side target is on
+
+def binary_search(arr, target):
```

Comment:
looks very nice! since you're checking for arr[midpoint], it makes great sense to set first & last to midpoint+1 and midpoint-1, respectively